backend/swift/swift.go

```diff
@@ -583,7 +588,9 @@ func (f *Fs) Mkdir(dir string) error {
 		_, _, err = f.c.Container(f.container)
 	}
 	if err == swift.ContainerNotFound {
-		err = f.c.ContainerCreate(f.container, nil)
+		headers := swift.Headers{}
+		headers["X-Storage-Policy"] = f.opt.StoragePolicy
```

Comment:
This will be adding a blank storage policy header when it isn't defined - I think it would be better not to supply the header at all in that case - what do you think?

Reply:
You're right, it's cleaner from an API point of view.
Although strictly seen it doesn't make any difference for swift, as an undefined header and empty string are handled identically.
https://github.com/openstack/swift/blob/86d78f6667392b2ee0ca017da8f54ff153aca491/swift/proxy/controllers/container.py#L61